src/main/java/io/camunda/testing/assertions/ProcessInstanceAssertions.java

```diff
@@ -388,4 +391,34 @@ public ProcessInstanceAssertions isNotWaitingForMessage(final String... messageN
             });
     return openMessageSubscriptions;
   }
+
+  public ProcessInstanceAssertions hasVariable(final String name) {
+    extractVariables()
+        .withFailMessage(
+            "Process with key %s does not contain variable with name %s",
+            actual.getProcessInstanceKey(), name)
+        .containsKey(name);
+    return this;
+  }
+
+  public ProcessInstanceAssertions hasVariableWithValue(final String name, final String value) {
+    extractVariables()
+        .withFailMessage(
+            "Process with key %s does not contain variable with name %s and value %s",
```

Comment:
:x: I think this error message is not very helpful. It covers two conditions:
* no variable with that name exists
* a variable does exist, but has a different value

For both I think we should offer dedicated error messages. 
In the first case we could print out the variable names we found (maybe it was just a typo)
In the second case we should print out the value we encountered.

Reply:
Good point!

What about this: "The variable x does not have the expected value. The value passed in 'rawValue' is internally converted to a JSON String that yields 'jsonEquivalent'. However, the actual value (as JSON String) is 'actualJsonValue'."

I know it's a bit tedious, but this conversion but has caused me a great deal of frustration in the past. So anything we can do here to help the user is well warranted, in my mind.